TestProfile fails

I'm trying to create a Debian package for `profile`. The test `TestProfile` fails during the build of the packages:

	go test -v -p 4 github.com/pkg/profile
    === RUN   TestProfile
    --- FAIL: TestProfile (1.75s)
	profile_test.go:174: default profile (cpu)
	profile_test.go:174: memory profile
	profile_test.go:174: memory profile (rate 2048)
	profile_test.go:174: double start
	profile_test.go:174: block profile
	profile_test.go:174: profile path
	profile_test.go:174: profile path error
	profile_test.go:194: stderr: wanted '[could not create initial output]', got '2017/05/09 06:50:47 profile: cpu profiling enabled, README.md/cpu.pprof
		2017/05/09 06:50:47 profile: cpu profiling disabled, README.md/cpu.pprof
		'
	profile_test.go:209: expected error
	profile_test.go:174: multiple profile sessions
	profile_test.go:174: profile quiet

What happens if you skip the -v and -p flags

On Tue, May 9, 2017 at 4:59 PM, legrostdg <notifications@github.com> wrote:

> I'm trying to create a Debian package for profile. The test TestProfile
> fails during the build of the packages:
>
> go test -v -p 4 github.com/pkg/profile
> === RUN   TestProfile
> --- FAIL: TestProfile (1.75s)
> profile_test.go:174: default profile (cpu)
> profile_test.go:174: memory profile
> profile_test.go:174: memory profile (rate 2048)
> profile_test.go:174: double start
> profile_test.go:174: block profile
> profile_test.go:174: profile path
> profile_test.go:174: profile path error
> profile_test.go:194: stderr: wanted '[could not create initial output]', got '2017/05/09 06:50:47 profile: cpu profiling enabled, README.md/cpu.pprof
> 	2017/05/09 06:50:47 profile: cpu profiling disabled, README.md/cpu.pprof
> 	'
> profile_test.go:209: expected error
> profile_test.go:174: multiple profile sessions
> profile_test.go:174: profile quiet
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/pkg/profile/issues/39>, or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAcA4O4sS5ymKLbvM_TTlN3WSkFrwFuks5r4A7cgaJpZM4NU5d4>
> .
>

go test github.com/pkg/profile
    --- FAIL: TestProfile (1.78s)
	profile_test.go:174: default profile (cpu)
	profile_test.go:174: memory profile
	profile_test.go:174: memory profile (rate 2048)
	profile_test.go:174: double start
	profile_test.go:174: block profile
	profile_test.go:174: profile path
	profile_test.go:174: profile path error
	profile_test.go:194: stderr: wanted '[could not create initial output]', got '2017/05/09 07:42:52 profile: cpu profiling enabled, README.md/cpu.pprof
		2017/05/09 07:42:52 profile: cpu profiling disabled, README.md/cpu.pprof
		'
	profile_test.go:209: expected error
	profile_test.go:174: multiple profile sessions
	profile_test.go:174: profile quiet
    FAIL
    FAIL	github.com/pkg/profile	1.784s

Ok, it was worth a shot, let me take a look at that test.

On Tue, 9 May 2017, 17:44 legrostdg <notifications@github.com> wrote:

> go test github.com/pkg/profile
> --- FAIL: TestProfile (1.78s)
> profile_test.go:174: default profile (cpu)
> profile_test.go:174: memory profile
> profile_test.go:174: memory profile (rate 2048)
> profile_test.go:174: double start
> profile_test.go:174: block profile
> profile_test.go:174: profile path
> profile_test.go:174: profile path error
> profile_test.go:194: stderr: wanted '[could not create initial output]', got '2017/05/09 07:42:52 profile: cpu profiling enabled, README.md/cpu.pprof
> 	2017/05/09 07:42:52 profile: cpu profiling disabled, README.md/cpu.pprof
> 	'
> profile_test.go:209: expected error
> profile_test.go:174: multiple profile sessions
> profile_test.go:174: profile quiet
> FAIL
> FAIL	github.com/pkg/profile	1.784s
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/pkg/profile/issues/39#issuecomment-300087360>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/AAAcA0CFLDsR-awJEK4BWsXn_h9-JfWLks5r4BlrgaJpZM4NU5d4>
> .
>

--- a/profile_test.go
+++ b/profile_test.go
@@ -14,13 +14,20 @@ import (
 
 type checkFn func(t *testing.T, stdout, stderr []byte, err error)
 
-var profileTests = []struct {
-	name   string
-	code   string
-	checks []checkFn
-}{{
-	name: "default profile (cpu)",
-	code: `
+func TestProfile(t *testing.T) {
+	f, err := ioutil.TempFile("", "profile_test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
+
+	var profileTests = []struct {
+		name   string
+		code   string
+		checks []checkFn
+	}{{
+		name: "default profile (cpu)",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -29,14 +36,14 @@ func main() {
 	defer profile.Start().Stop()
 }	
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("profile: cpu profiling enabled"),
-		NoErr,
-	},
-}, {
-	name: "memory profile",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("profile: cpu profiling enabled"),
+			NoErr,
+		},
+	}, {
+		name: "memory profile",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -45,14 +52,14 @@ func main() {
 	defer profile.Start(profile.MemProfile).Stop()
 }	
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("profile: memory profiling enabled"),
-		NoErr,
-	},
-}, {
-	name: "memory profile (rate 2048)",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("profile: memory profiling enabled"),
+			NoErr,
+		},
+	}, {
+		name: "memory profile (rate 2048)",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -61,14 +68,14 @@ func main() {
 	defer profile.Start(profile.MemProfileRate(2048)).Stop()
 }	
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("profile: memory profiling enabled (rate 2048)"),
-		NoErr,
-	},
-}, {
-	name: "double start",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("profile: memory profiling enabled (rate 2048)"),
+			NoErr,
+		},
+	}, {
+		name: "double start",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -78,14 +85,14 @@ func main() {
 	profile.Start()
 }	
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("cpu profiling enabled", "profile: Start() already called"),
-		Err,
-	},
-}, {
-	name: "block profile",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("cpu profiling enabled", "profile: Start() already called"),
+			Err,
+		},
+	}, {
+		name: "block profile",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -94,14 +101,14 @@ func main() {
 	defer profile.Start(profile.BlockProfile).Stop()
 }	
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("profile: block profiling enabled"),
-		NoErr,
-	},
-}, {
-	name: "mutex profile",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("profile: block profiling enabled"),
+			NoErr,
+		},
+	}, {
+		name: "mutex profile",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -110,14 +117,14 @@ func main() {
 	defer profile.Start(profile.MutexProfile).Stop()
 }
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("profile: mutex profiling enabled"),
-		NoErr,
-	},
-}, {
-	name: "profile path",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("profile: mutex profiling enabled"),
+			NoErr,
+		},
+	}, {
+		name: "profile path",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -126,30 +133,30 @@ func main() {
 	defer profile.Start(profile.ProfilePath(".")).Stop()
 }	
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("profile: cpu profiling enabled, cpu.pprof"),
-		NoErr,
-	},
-}, {
-	name: "profile path error",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("profile: cpu profiling enabled, cpu.pprof"),
+			NoErr,
+		},
+	}, {
+		name: "profile path error",
+		code: `
 package main
 
 import "github.com/pkg/profile"
 
 func main() {
-		defer profile.Start(profile.ProfilePath("README.md")).Stop()
+		defer profile.Start(profile.ProfilePath("` + f.Name() + `")).Stop()
 }	
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("could not create initial output"),
-		Err,
-	},
-}, {
-	name: "multiple profile sessions",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("could not create initial output"),
+			Err,
+		},
+	}, {
+		name: "multiple profile sessions",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -162,23 +169,23 @@ func main() {
 	profile.Start(profile.MutexProfile).Stop()
 }
 `,
-	checks: []checkFn{
-		NoStdout,
-		Stderr("profile: cpu profiling enabled",
-			"profile: cpu profiling disabled",
-			"profile: memory profiling enabled",
-			"profile: memory profiling disabled",
-			"profile: block profiling enabled",
-			"profile: block profiling disabled",
-			"profile: cpu profiling enabled",
-			"profile: cpu profiling disabled",
-			"profile: mutex profiling enabled",
-			"profile: mutex profiling disabled"),
-		NoErr,
-	},
-}, {
-	name: "profile quiet",
-	code: `
+		checks: []checkFn{
+			NoStdout,
+			Stderr("profile: cpu profiling enabled",
+				"profile: cpu profiling disabled",
+				"profile: memory profiling enabled",
+				"profile: memory profiling disabled",
+				"profile: block profiling enabled",
+				"profile: block profiling disabled",
+				"profile: cpu profiling enabled",
+				"profile: cpu profiling disabled",
+				"profile: mutex profiling enabled",
+				"profile: mutex profiling disabled"),
+			NoErr,
+		},
+	}, {
+		name: "profile quiet",
+		code: `
 package main
 
 import "github.com/pkg/profile"
@@ -187,10 +194,8 @@ func main() {
         defer profile.Start(profile.Quiet).Stop()
 }       
 `,
-	checks: []checkFn{NoStdout, NoStderr, NoErr},
-}}
-
-func TestProfile(t *testing.T) {
+		checks: []checkFn{NoStdout, NoStderr, NoErr},
+	}}
 	for _, tt := range profileTests {
 		t.Log(tt.name)
 		stdout, stderr, err := runTest(t, tt.code)

